subsys/net/lib/http_server/src/server.c

```diff
@@ -0,0 +1,538 @@
+	print_http2_frames(frames, frame_count);
+
+	readBytes = recv(ctx->client_fds[i].fd, frame, sizeof(frame), 0);
+	printf("readBytes length: %u\n", readBytes);
+	if (readBytes < 0) {
+	    perror("ERROR reading from socket");
+	    close_client_connection(ctx, i);
+	} else if (readBytes == 0) {
+	    printf("Connection closed by peer.\n");
+	}
+	frame_count = parse_http2_frames(frame, readBytes, frames);
+	print_http2_frames(frames, frame_count);
+    }
+
+    /* Send a SETTINGS frame after receiving a valid preface */
+    ssize_t n =
+	sendall(ctx->client_fds[i].fd, settings_frame, sizeof(settings_frame));
+    if (n < 0) {
+	perror("ERROR writing to socket");
+	exit(EXIT_FAILURE);
+    }
+
+    n = sendall(ctx->client_fds[i].fd, settings_ack, sizeof(settings_ack));
+    if (n < 0) {
+	perror("ERROR writing to socket");
+	exit(EXIT_FAILURE);
+    }
+
+    /* Read the header of the next frame */
+    readBytes = recv(ctx->client_fds[i].fd, frame, sizeof(frame), 0);
+    if (readBytes < 0) {
+	perror("ERROR reading from socket");
+	close_client_connection(ctx, i);
+    } else if (readBytes == 0) {
+	printf("Connection closed by peer.\n");
+    }
+    frame_count = parse_http2_frames(frame, readBytes, frames);
+    print_http2_frames(frames, frame_count);
+
+    n = sendall(ctx->client_fds[i].fd, settings_frame, sizeof(settings_frame));
+    if (n < 0) {
+	perror("ERROR writing to socket");
+	exit(EXIT_FAILURE);
+    }
+
+    unsigned char response_headers_frame[11];
+    generate_response_headers_frame(response_headers_frame, stream_header_id);
+
+    n = sendall(ctx->client_fds[i].fd, response_headers_frame,
+		sizeof(response_headers_frame));
+    if (n < 0) {
+	perror("ERROR writing to socket");
+	exit(EXIT_FAILURE);
+    }
+
+    readBytes = recv(ctx->client_fds[i].fd, frame, sizeof(frame), 0);
+    if (readBytes < 0) {
+	perror("ERROR reading from socket");
+	close_client_connection(ctx, i);
+    } else if (readBytes == 0) {
+	printf("Connection closed by peer.\n");
+    }
+
+    frame_count = parse_http2_frames(frame, readBytes, frames);
+    print_http2_frames(frames, frame_count);
+
+    size_t content_size = sizeof(content2);
+    sendData(ctx->client_fds[i].fd, content2, content_size, 0x00, 0x01, 0x01);
+
+    readBytes = recv(ctx->client_fds[i].fd, frame, sizeof(frame), 0);
+    if (readBytes < 0) {
+	perror("ERROR reading from socket");
+	close_client_connection(ctx, i);
+    } else if (readBytes == 0) {
+	printf("Connection closed by peer.\n");
+    }
+
+    frame_count = parse_http2_frames(frame, readBytes, frames);
+    print_http2_frames(frames, frame_count);
+
+    hasUpgradeHeader = 1;
+    close_client_connection(ctx, i);
+}
+
+int http2_server_init(struct http2_server_ctx *ctx,
+		      struct http2_server_config *config)
+{
+    /* Create a socket */
+    ctx->sockfd = socket(config->address_family, SOCK_STREAM, 0);
+    if (ctx->sockfd < 0) {
+	perror("socket");
+	exit(EXIT_FAILURE);
```

Comment:
We should not use `exit()` in the library code - we don't want to break the system in case of simple errors. Just return a negative error code here (`return -errno`).